runtime/testy/Ansible.py

```diff
@@ -167,11 +198,25 @@ def unpackageData(self):
         function is currently unimplemented.
         """
         def unpackage(data):
-            """Function that takes a packaged proto and unpackages item
+            """Function that takes a packaged proto and unpackages the item
 
-            Currently simply returns the original data. Needs to be implemented
+            Parses through the python pseudo-class created by the protobuf and stores it into a dictionary.
+            All of the axes data and the button data, enumerates each value to follow a mapping shared by dawn,
+            and stores it in the dictionary with the gamepad index as a key.
+
+            student code status is also stored in this dictionary. This dictionary is added to the overall state
+            through the update method implemented in state manager.
             """
-            return data 
+            unpackaged_data = {}
+            received_proto = ansible_pb2.DawnData()
+            received_proto.ParseFromString(data)
+            unpackaged_data["student_code_status"] = received_proto.student_code_status
+            for gamepad in received_proto.gamepads:
+                gamepad_dict = {}
+                gamepad_dict["axes"] = {i : axis for i, axis in zip(range(0,4), gamepad.axes)}
```

Comment:
Use `enumerate(gamepad.axes)` instead of `zip`.

Moreover, the entire line could be `gamepad_dict["axes"] = dict(enumerate(gamepad.axes))`.

But what's even less clear is why the gamepads are stored in a dict as opposed to a list in the first place. Could you explain?